src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

```diff
@@ -552,9 +555,50 @@ internal override void AfterAddingTypeMembersChecks(ConversionsBase conversions,
             }
 
             diagnostics.Add(Location, useSiteInfo);
+
+            if (IsPartialDefinition && OtherPartOfPartial is { } implementation)
+            {
+                PartialPropertyChecks(implementation, diagnostics);
+            }
+        }
+
+        private void PartialPropertyChecks(SourcePropertySymbol implementation, BindingDiagnosticBag diagnostics)
+        {
+            Debug.Assert(this.IsPartialDefinition);
+            Debug.Assert((object)this != implementation);
+            Debug.Assert((object?)this.OtherPartOfPartial == implementation);
+
+            // PROTOTYPE(partial-properties): check and diagnose all disallowed differences between parts.
+            // PROTOTYPE(partial-properties): we should likely report a warning when only one of the parts has an explicit access modifier for consistency with methods.
+            if (DeclaredAccessibility != implementation.DeclaredAccessibility)
+            {
+                // PROTOTYPE(partial-properties): we should either generalize the existing diagnostic or introduce new set of diagnostics.
+                diagnostics.Add(ErrorCode.ERR_PartialMethodAccessibilityDifference, implementation.GetFirstLocation());
+            }
         }
 
         private static BaseParameterListSyntax? GetParameterListSyntax(CSharpSyntaxNode syntax)
             => (syntax as IndexerDeclarationSyntax)?.ParameterList;
+
+        internal bool IsPartial => (_modifiers & DeclarationModifiers.Partial) != 0;
+
+        internal SourcePropertySymbol? OtherPartOfPartial { get => _otherPartOfPartial; }
+
+        internal bool IsPartialDefinition => IsPartial && !AccessorsHaveImplementation && !IsExtern;
+
+        internal bool IsPartialImplementation => IsPartial && (AccessorsHaveImplementation || IsExtern);
+
+        internal static void InitializePartialPropertyParts(SourcePropertySymbol definition, SourcePropertySymbol implementation)
+        {
+            Debug.Assert(definition.IsPartialDefinition);
+            Debug.Assert(implementation.IsPartialImplementation);
+
+            Debug.Assert(definition._otherPartOfPartial is null || definition._otherPartOfPartial == implementation);
```

Comment:
When do we fall into the alternative case of this assertion (or the one below)?

Reply:
Threads can race to perform this work. That case indicates another thread already set the `_otherPartOfPartial`. There is a similar check being performed in the corresponding `InitializePartialMethodParts`.

https://github.com/dotnet/roslyn/blob/28b39facc518448c7af332286e847cefeb683d90/src/Compilers/CSharp/Portable/Symbols/Source/SourceOrdinaryMethodSymbol.cs#L998-L1008

Reply:
Because this race, the assertion here itself is actually racy and may not catch issues. To avoid racing, we'll want to read the `_otherPartOfPartial` fields only once. I'd recommend:

```cs
definition._otherPartOfPartial is not { } otherPart || otherPart == implementation
```

We'll also want to assert, after the sets, that the `_otherPartOfPartial`s are reference equal to what we set, otherwise we could race there too.

Reply:
The alternative is to use an interlockedcompareexchange and assert after the ice.